working/4200-metaobjects/feature-specification.md

```diff
@@ -0,0 +1,519 @@
+forwarding method is `R m<typeParms>(parms) => C.m<typeArgs>(args);`, where
+`typeArgs` is a comma separated list of the type variables declared in
+`typeParms` in declaration order, and `args` is a comma separated list of
+identifiers declared as positional parameters in `parms`, followed by
+actual arguments of the form `id: id` for the named parameters in `parms`.
+
+In the case where a default value resolves to a declaration in `C` and does
+not have the prefix `C.`, this prefix is added when the corresponding
+default value is specified in the metaobject class. *For example, 
+`int i = myDefault` in `C` will be `int i = C.myDefault` in the derived
+forwarding declaration when `myDefault` is declared in `C`.*
+
+If `C(parms) ...` or `factory C(parms) ...` is a constructor declared by `C`
+then the corresponding forwarding declaration is
+`C<X1 .. Xk> call(parms1) => C(args);`, here `args` are derived from `parms1`
+in the same way as with the static method above, and `parms1` is derived
+from `parms` by erasing `this.` and `super.` and inserting the inferred
+types and default values that are implied by `this.` and `super.` in the
+cases where those declared types and default values have been omitted.
+
+If `C.name(parms) ...` or `factory C.name(parms) ...` is a constructor
+declared by `C` then the corresponding forwarding declaration is
+`C<X1 .. Xk> name(parms1) => C.name(args);`, here `args` are derived from
+`parms1` in the same way as with the static method above, and `parms1` is
+derived from `parms` in the same way as in the previous case.
+
+### Dynamic Semantics
+
+It is allowed, but not required, for every metaobject to be constant if
+possible. *Hence, programs should not depend on the identity of
+metaobjects.*
+
+When invoked on an object whose run-time type is `C<T1 .. Tk>`, in the case
+where `C` has a `static implements` or `static extends` clause, the
+implementation of the getter `runtimeType` in `Object` returns an instance
+of the metaobject class for `C` with the same type arguments `T1 .. Tk`.
+
+*For example, `C<int, String>().runtimeType` returns `MetaC<int, String>()`
+if `C` has the clause `static implements SomeInterface` and `MetaC` is the
+implicitly induced metaobject class for `C`.*
+
+In the case where a type `T` is introduced by a declaration that has a
+`static implements` or a `static extends` clause, the step whereby this
+type is evaluated as an expression yields the corresponding metaobject,
+that is, an instance of the metaobject class for `T`, passing the same
+actual type arguments as the ones which are part of `T`.
+
+*For example, if we are evaluating a type parameter `X` as an expression,
+and the value of `X` is a type `C<S1, S2>` that has a metaobject class
+`MetaC` then the result will be an instance of `MetaC<S1, S2>`.*
+
+The metaobject has standard semantics, everything follows the normal rules
+of Dart based on the metaobject class declaration.
+
+### Changelog
```

Comment:
Maybe include the changes of the version 1.1?

Reply:
Why is the version in the header marked as 1.1, then? I don't know if I understand.

Reply:
Oops, that's a typo. Fixed.